pkg/stack/stack.go

```diff
@@ -94,6 +97,21 @@ func (s *Stack) GetImageBuilds() (map[string]riov1.ImageBuild, error) {
 				}
 
 				buildConfig[svc.Name] = *svc.Spec.Build
+			} else {
+				// use docker reference parser to determine if its a valid image reference
+				_, err = reference.ParseAnyReference(svc.Spec.Image)
+				if err != nil { //fallback and see if its directory
+					fileInfo, err := os.Stat(svc.Spec.Image)
+					if err != nil {
+						return nil, errors.Wrap(err, "error parsing image field")
+					}
+					if fileInfo.IsDir() {
+						buildConfig[svc.Name] = riov1.ImageBuild{BuildContext: svc.Spec.Image}
+						svc.Spec.Image = ""
+					} else {
+						return nil, errors.Wrap(err, "image field is not a directory")
```

Comment:
Is there a reason it has to be a dir, can't I just have 5 named dockerfiles in this dir ? `Dockerfile.agent`, `Dockerfile.web`, `Dockerfile.db`  ...

Reply:
This sounds like a design question and my answer is leaning towards no. 
1. If you give us a directory where we should have had an `image` then
2. We will do some _minor_ inferencing and try to build the Dockerfile we find in there. 
3. We are mimicking the behavior of Dockerbuild for a simple flow https://docs.docker.com/engine/reference/commandline/build/#build-with-path
- More advanced flows should be defining multiple images within the Riofile construct

Reply:
Yea Riofile should work for now or they can just make dirs. If someone requests we can make a future enhancement.

Reply:
Dockerfile name can be passed in build args though. We have `Dockerfile` field to specify the name of Dockerfile. It defaults to `Dockerfile`.